pkg/export/disk/disk.go

```diff
@@ -31,24 +33,62 @@ type Connection struct {
 	currentAuditRun string
 }
 
+// FailedConnection wraps a Connection with retry metadata.
+type FailedConnection struct {
+	Connection
+	FailedAt    time.Time
+	RetryCount  int
+	NextRetryAt time.Time
+}
+
 type Writer struct {
-	openConnections map[string]Connection
+	mu                           sync.RWMutex
+	openConnections              map[string]Connection
+	closedConnections            map[string]FailedConnection
+	cleanupDone                  chan struct{}
+	cleanupOnce                  sync.Once
+	closeAndRemoveFilesWithRetry func(conn Connection) error
 }
 
 const (
 	Name                = "disk"
 	maxAllowedAuditRuns = 5
 	maxAuditResults     = "maxAuditResults"
 	violationPath       = "path"
+	// Connection cleanup constants.
+	cleanupInterval  = 5 * time.Minute
+	maxRetryAttempts = 10
+	maxConnectionAge = 30 * time.Minute
+	baseRetryDelay   = 15 * time.Second
 )
 
 var Connections = &Writer{
-	openConnections: make(map[string]Connection),
+	openConnections:   make(map[string]Connection),
+	closedConnections: make(map[string]FailedConnection),
+	cleanupDone:       make(chan struct{}),
+	closeAndRemoveFilesWithRetry: func(conn Connection) error {
+		return wait.ExponentialBackoff(retry.DefaultBackoff, func() (bool, error) {
+			if conn.File != nil {
+				if err := conn.unlockAndCloseFile(); err != nil {
+					return false, fmt.Errorf("error closing file: %w", err)
+				}
+			}
+			if err := os.RemoveAll(conn.Path); err != nil {
+				return false, fmt.Errorf("error deleting violations stored at old path: %w", err)
+			}
+			return true, nil
+		})
+	},
 }
 
 var log = logf.Log.WithName("disk-driver").WithValues(logging.Process, "export")
 
 func (r *Writer) CreateConnection(_ context.Context, connectionName string, config interface{}) error {
+	// Start cleanup goroutine on first connection
+	r.cleanupOnce.Do(func() {
```

Comment:
where do you `close(r.cleanupDone)`?

Reply:
where do you close it? I cant find it.

Reply:
Here - https://github.com/open-policy-agent/gatekeeper/pull/3946/files#diff-1de204e08589a6e8a84146c2510cb29443d26e835b1b5b77ba3970ba2fe5a7faR418